api/src/main/java/io/smallrye/reactive/messaging/annotations/EmitterFactoryFor.java

```diff
@@ -29,4 +31,27 @@
      * @return the emitter interface
      */
     Class<?> value();
+
+    final class EmitterFactoryForLiteral extends AnnotationLiteral<EmitterFactoryFor> implements EmitterFactoryFor {
```

Comment:
Nitpicking: the literal class, when nested in the annotation type, is by convention called just `Literal`.